sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ComplexTypeSuite.scala

```diff
@@ -117,6 +117,22 @@ class ComplexTypeSuite extends SparkFunSuite with ExpressionEvalHelper {
     checkEvaluation(getArrayStructFields(nullArrayStruct, "a"), null)
   }
 
+  test("CreateArray") {
+    val intSeq = Seq(5, 10, 15, 20, 25)
+    val longSeq = intSeq.map(_.toLong)
+    val strSeq = intSeq.map(_.toString)
+    checkEvaluation(CreateArray(intSeq.map(Literal(_))), intSeq, EmptyRow)
+    checkEvaluation(CreateArray(longSeq.map(Literal(_))), longSeq, EmptyRow)
+    checkEvaluation(CreateArray(strSeq.map(Literal(_))), strSeq, EmptyRow)
+
+    val intWithNull = intSeq.map(Literal(_)).+:(Literal.create(null, IntegerType))
+    val longWithNull = longSeq.map(Literal(_)).+:(Literal.create(null, LongType))
+    val strWithNull = strSeq.map(Literal(_)).+:(Literal.create(null, StringType))
+    checkEvaluation(CreateArray(intWithNull), intSeq.+:(null), EmptyRow)
+    checkEvaluation(CreateArray(longWithNull), longSeq.+:(null), EmptyRow)
+    checkEvaluation(CreateArray(strWithNull), strSeq.+:(null), EmptyRow)
```

Comment:
Sorry to nit-pick, but what is happening here is kind of confusing:

```scala
scala> List(1,2).+:(null)
res0: List[Any] = List(null, 1, 2)
```

Perhaps just use infix notation for operators: 
```scala
scala> List(1,2) :+ null
res1: List[Any] = List(1, 2, null)
```